File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:

        left_sums = []
        c_sum = 0

        for num in nums:
            left_sums.append(c_sum)
            c_sum += num
        
        right_sums = [0] * len(nums)
        c_sum = 0

        for i in range(len(nums) - 1, -1, -1):
            right_sums[i] = c_sum
            c_sum += nums[i]
        
        for i in range(len(nums)):
            if left_sums[i] == right_sums[i]:
                return i
        
        return -1
        

File: 0724-find-pivot-index/0724-find-pivot-index.cs
public class Solution {
    public int PivotIndex(int[] nums) {
        var leftSum = new int[nums.Length];
        var rightSum = new int[nums.Length];
        int curr = 0;
        for (int i = nums.Length - 1; i >= 0; i--) {
            rightSum[i] = curr;
            curr += nums[i];
        }
        curr = 0;
        for (int i = 0; i < nums.Length; i++) {
            leftSum[i] = curr;
            if (leftSum[i] == rightSum[i]) return i;
            curr += nums[i];
        }
        return -1;

    }
}